apollo-federation/src/operation/selection_map.rs

```diff
@@ -196,6 +199,14 @@ impl PartialEq for SelectionMap {
 
 impl Eq for SelectionMap {}
 
+impl Hash for SelectionMap {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.values()
+            .sorted()
```

Comment:
Note this does allocate a vec for all the selections. It seems like a necessary evil though.

Reply:
yes, this is to allow us hash things in order independent way (so the contract with partial eq holds)